k8s-operator/apis/v1alpha1/types_connector.go

```diff
@@ -41,40 +46,70 @@ type ConnectorList struct {
 	Items []Connector `json:"items"`
 }
 
-// ConnectorSpec defines the desired state of a ConnectorSpec.
+// ConnectorSpec describes a Tailscale node to be deployed in the cluster.
+// +kubebuilder:validation:XValidation:rule="has(self.subnetRouter) || self.exitNode == true",message="A Connector needs to be either an exit node or a subnet router, or both."
 type ConnectorSpec struct {
-	// SubnetRouter configures a Tailscale subnet router to be deployed in
-	// the cluster. If unset no subnet router will be deployed.
+	// Tags that the Tailscale node will be tagged with.
+	// Defaults to [tag:k8s].
+	// To autoapprove the subnet routes or exit node defined by a Connector,
+	// you can configure Tailscale ACLs to give these tags the necessary
+	// permissions.
+	// See https://tailscale.com/kb/1018/acls/#auto-approvers-for-routes-and-exit-nodes
```

Comment:
> Since this seems to be concatenated into a long string when tailscale.com_connectors.yaml is generated.

That doesn't actually matter, in general no-one looks at generated CRD yamls, usually you'd use something like `kubectl explain` which wraps it:

```
k explain connector.spec
GROUP:      tailscale.com
KIND:       Connector
VERSION:    v1alpha1

FIELD: spec <Object>

DESCRIPTION:
    ConnectorSpec describes the desired Tailscale component.
    
FIELDS:
  exitNode      <boolean>
    IsExitNode defines whether the Connector node should act as a Tailscale exit
    node. Defaults to false. https://tailscale.com/kb/1103/exit-nodes

  hostname      <string>
    Hostname is the tailnet hostname that should be assigned to the Connector
    node. If unset, hostname is defaulted to <connector name>-connector.
    Hostname can contain lower case letters, numbers and dashes, it must not
    start or end with a dash and must be between 2 and 63 characters long.

  subnetRouter  <Object>
    SubnetRouter defines subnet routes that the Connector node should expose to
    tailnet. If unset, none are exposed. https://tailscale.com/kb/1019/subnets/

  tags  <[]string>
    Tags that the Tailscale node will be tagged with. Defaults to [tag:k8s]. To
    autoapprove the subnet routes or exit node defined by a Connector, you can
    configure Tailscale ACLs to give these tags the necessary permissions. See
    https://tailscale.com/kb/1018/acls/#auto-approvers-for-routes-and-exit-nodes
    If you specify custom tags here, you must also make the operator an owner of
    these tags. See
    https://tailscale.com/kb/1236/kubernetes-operator/#setting-up-the-kubernetes-operator.
    Tags cannot be changed once a Connector node has been created. Tag values
    must be in form ^tag:[a-zA-Z][a-zA-Z0-9-]*$.
```

(But yes, it does need a period 👍🏼 )